radix-engine/src/kernel/call_frame.rs

```diff
@@ -62,12 +64,24 @@ pub enum StableReferenceType {
     DirectAccess,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TransientReference {
+    ref_count: usize,
+    root_node_type: RootNodeType,
```

Comment:
This is probably a naming thing, but `RootNodeType` abstraction doesn't look right, and I think we're mixing reference type with node location.